app/models/comment.php

```diff
@@ -0,0 +1,50 @@
+<?php
+class Comment extends AppModel
+{
+    public $validation = array(
+        'body' => array(
+            'length' => array(
+                'validate_between', 1, 200,
+                ),
+            ),
+        );
+
+    public function getAll($offset, $limit)
+    {
+        $comments = array();
+        $thread = Param::get('thread_id');
+        $db = DB::conn();
+
+        $rows = $db->rows(
+        "SELECT * FROM comment WHERE thread_id = ? ORDER BY created LIMIT {$offset},{$limit}", array($thread));
+
+        foreach ($rows as $row) {
+            $comments[] = new self($row);
+        }
+
+        return $comments;
+    }
+
+    public static function countAll()
+    {
+        $db = DB::conn();
+        $id = Param::get('thread_id');
+        return $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = {$id}");
```

Comment:
please use placeholder po for $id instead of directly using the param in your query :bow: